lib/stripe/stripe_client.rb

```diff
@@ -14,28 +14,29 @@ class StripeClient
     @last_connection_manager_gc = Util.monotonic_time
 
     # Initializes a new StripeClient
-    def initialize(config_overrides = {})
+    def initialize(config_arg = {})
```

Comment:
Still supports the same types (i.e. hash, string, etc.) before, but I've expanded it so that you can optionally inject a full configuration object. The reason is that it allows me to initialize all default clients with the same global configuration object (`Stripe.configuration`). That means that when global configuration changes (i.e. `Stripe.write_timeout = 3.0`), that new configuration propagates to every default client.

Reply:
This is fantastic and feels much more consistent! When I was writing specs, I found that I wanted to pass a `StripeConfiguration` object but had to remind myself to use a Hash.

Reply:
Perfect. Thanks!